src/diffusers/models/auto_model.py

```diff
@@ -156,12 +159,28 @@ def from_pretrained(cls, pretrained_model_or_path: Optional[Union[str, os.PathLi
             "subfolder": subfolder,
         }
 
-        config = cls.load_config(pretrained_model_or_path, **load_config_kwargs)
-        orig_class_name = config["_class_name"]
+        try:
+            mindex_kwargs = {k: v for k, v in load_config_kwargs.items() if k != "subfolder"}
+            mindex_kwargs["filename"] = "model_index.json"
+            config_path = hf_hub_download(pretrained_model_or_path, **mindex_kwargs)
+            config = cls.load_config(config_path, **load_config_kwargs)
+            library, orig_class_name = config[subfolder]
+            model_cls, _ = get_class_obj_and_candidates(
+                library_name=library,
+                class_name=orig_class_name,
+                importable_classes=ALL_IMPORTABLE_CLASSES,
+                pipelines=pipelines,
+                is_pipeline_module=hasattr(pipelines, library),
+                component_name=subfolder,
+                cache_dir=constants.HF_HUB_CACHE,
```

Comment:
```suggestion
                cache_dir=None,
```
let's jjust pass None here since we ddon't actually have a cached_dir so not very meaningful
and update the other function so it works with cache_dir=None and component_name=NOne